crates/evm/core/src/utils.rs

```diff
@@ -203,19 +213,31 @@ pub fn create2_handler_register<I: InspectorExt>(
                 .push((ctx.evm.journaled_state.depth(), call_inputs.clone()));
 
             // Sanity check that CREATE2 deployer exists.
-            // We check which deployer we are using to separate the logic for zkSync and original
-            // foundry.
-            let mut code_hash = ctx.evm.load_account(DEFAULT_CREATE2_DEPLOYER)?.info.code_hash;
-
+            // NOTE(zk): made mut to apply later check
+            let mut code_hash = ctx.evm.load_account(create2_deployer)?.info.code_hash;
+            // NOTE(zk): We check which deployer we are using to separate the logic for zkSync
+            // and original foundry.
+            // TODO(zk): adding this check to skip comparing to evm create2 deployer
+            // hash, should we compare vs zkevm one?
+            let mut zk_is_create2_deployer = false;
             if call_inputs.target_address == DEFAULT_CREATE2_DEPLOYER_ZKSYNC {
                 code_hash = ctx.evm.load_account(call_inputs.target_address)?.info.code_hash;
-            };
-
+                zk_is_create2_deployer = true;
```

Comment:
Maybe it makes sense to assign `code_hash` to `DEFAULT_CREATE2_DEPLOYER_CODEHASH`. But also good to have a clear zksync context.